Haskell

thenaesh opened the pull request.

thenaesh committed:
First draft

Signed-off-by: Thenaesh Elango <thenaeshelango@gmail.com>

Neurrone commented on contents/haskell/Haskell.md:
```diff
@@ -0,0 +1,195 @@
+
+Haskell is a purely functional programming language with strong, static,
+inferred typing. It is one of the languages that feature the
+[second-order lambda calculus](https://en.wikipedia.org/wiki/System_F)
```
Don't think a newcomer to the language will appreciate the significance of that, or be able to understand the wikipedia page.

thenaesh replied:
Noted. Removing.

Neurrone commented on contents/haskell/Haskell.md:
```diff
@@ -0,0 +1,195 @@
+side-effect-free computation makes it a valuable asset in software engineering
+contexts. Programs written in Haskell tend to be easy to test, refactor and
+debug, with the compiler usually catching all bugs before the program can even
+be compiled and run. Consequently, Haskell codebases are extraordinarily stable.
```
Any links to demonstrate this?

thenaesh replied:
Well, no concrete examples (I don't think people go around actually documenting occurrences of this), but lots of pages explaining this. I'll throw one in here.

Neurrone commented on contents/haskell/Haskell.md:
```diff
@@ -0,0 +1,195 @@
+* MacOS
+  * Homebrew
+  * ordinary installer
+* Windows
```
Might not be necessary to list out all the OSes supported.

Neurrone commented on contents/haskell/Haskell.md:
```diff
@@ -0,0 +1,195 @@
+Stack is a package manager of sorts for Haskell, similar to NPM. Installation
+instructions may be found in the
+[Stack Documentation](https://docs.haskellstack.org/en/stable/README/), and is
+fairly standard.
```
For someone just learning the language, is stack recommended over system-wide install?

thenaesh replied:
No. But I put it in there because the moment the learner wants to create an actual project, Stack comes in handy.

damithc commented:
@thenaesh let me know after approved by peer reviewer

thenaesh committed:
Perform recommended edits

The part about System F has been removed, and the part explaining
where to use a system-wide installation of Haskell as opposed to
using Stack has been moved to the start of the section regarding
installation. The list of OSes has been removed.

Signed-off-by: Thenaesh Elango <thenaeshelango@gmail.com>

thenaesh commented:
@Neurrone I've made a few edits. For your review.

damithc reviewed (commented):
@thenaesh This doc seems to be incomplete. Try to complete it by adding at least a short description for missing sections.  A quick overview of what it is and then a few links to learn more is good enough. I'll be merging this at the end of week 12.

thenaesh committed:
Add section on types

thenaesh committed:
add rest of book

thenaesh committed:
add comment on modules in Haskell

thenaesh commented:
@damithc The chapter is complete.

damithc committed:
Merge branch 'master' into haskell
